File: app/src/main/java/eu/toma/dev/playground/warcraft/model/item/Item.java
package eu.toma.dev.playground.warcraft.model.item;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Item {

    @SerializedName("id")
    @Expose
    public Integer id;
    @SerializedName("description")
    @Expose
    public String description;
    @SerializedName("name")
    @Expose
    public String name;
    @SerializedName("icon")
    @Expose
    public String icon;
    @SerializedName("stackable")
    @Expose
    public Integer stackable;
    @SerializedName("itemBind")
    @Expose
    public Integer itemBind;
    @SerializedName("bonusStats")
    @Expose
    public List<Object> bonusStats = null;
    @SerializedName("itemSpells")
    @Expose
    public List<ItemSpell> itemSpells = null;
    @SerializedName("buyPrice")
    @Expose
    public Integer buyPrice;
    @SerializedName("itemClass")
    @Expose
    public Integer itemClass;
    @SerializedName("itemSubClass")
    @Expose
    public Integer itemSubClass;
    @SerializedName("containerSlots")
    @Expose
    public Integer containerSlots;
    @SerializedName("inventoryType")
    @Expose
    public Integer inventoryType;
    @SerializedName("equippable")
    @Expose
    public Boolean equippable;
    @SerializedName("itemLevel")
    @Expose
    public Integer itemLevel;
    @SerializedName("maxCount")
    @Expose
    public Integer maxCount;
    @SerializedName("maxDurability")
    @Expose
    public Integer maxDurability;
    @SerializedName("minFactionId")
    @Expose
    public Integer minFactionId;
    @SerializedName("minReputation")
    @Expose
    public Integer minReputation;
    @SerializedName("quality")
    @Expose
    public Integer quality;
    @SerializedName("sellPrice")
    @Expose
    public Integer sellPrice;
    @SerializedName("requiredSkill")
    @Expose
    public Integer requiredSkill;
    @SerializedName("requiredAbility")
    @Expose
    public RequiredAbility requiredAbility;
    @SerializedName("requiredLevel")
    @Expose
    public Integer requiredLevel;
    @SerializedName("requiredSkillRank")
    @Expose
    public Integer requiredSkillRank;
    @SerializedName("itemSource")
    @Expose
    public ItemSource itemSource;
    @SerializedName("baseArmor")
    @Expose
    public Integer baseArmor;
    @SerializedName("hasSockets")
    @Expose
    public Boolean hasSockets;
    @SerializedName("isAuctionable")
    @Expose
    public Boolean isAuctionable;
    @SerializedName("armor")
    @Expose
    public Integer armor;
    @SerializedName("displayInfoId")
    @Expose
    public Integer displayInfoId;
    @SerializedName("nameDescription")
    @Expose
    public String nameDescription;
    @SerializedName("nameDescriptionColor")
    @Expose
    public String nameDescriptionColor;
    @SerializedName("upgradable")
    @Expose
    public Boolean upgradable;
    @SerializedName("heroicTooltip")
    @Expose
    public Boolean heroicTooltip;
    @SerializedName("context")
    @Expose
    public String context;
    @SerializedName("bonusLists")
    @Expose
    public List<Object> bonusLists = null;
    @SerializedName("availableContexts")
    @Expose
    public List<String> availableContexts = null;
    @SerializedName("bonusSummary")
    @Expose
    public BonusSummary bonusSummary;
    @SerializedName("artifactId")
    @Expose
    public Integer artifactId;

}

File: app/src/main/kotlin/warcraft/model/item/Item.kt
package warcraft.model.item

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Item {

    @SerializedName("id")
    @Expose
    var id: Int? = null

    @SerializedName("description")
    @Expose
    lateinit var description: String

    @SerializedName("name")
    @Expose
    lateinit var name: String

    @SerializedName("icon")
    @Expose
    lateinit var icon: String

    @SerializedName("stackable")
    @Expose
    var stackable: Int? = null

    @SerializedName("itemBind")
    @Expose
    var itemBind: Int? = null
    @SerializedName("bonusStats")
    @Expose
    var bonusStats: List<Any>? = null
    @SerializedName("itemSpells")
    @Expose
    var itemSpells: List<ItemSpell>? = null
    @SerializedName("buyPrice")
    @Expose
    var buyPrice: Int? = null
    @SerializedName("itemClass")
    @Expose
    var itemClass: Int? = null
    @SerializedName("itemSubClass")
    @Expose
    var itemSubClass: Int? = null
    @SerializedName("containerSlots")
    @Expose
    var containerSlots: Int? = null
    @SerializedName("inventoryType")
    @Expose
    var inventoryType: Int? = null
    @SerializedName("equippable")
    @Expose
    var equippable: Boolean? = null
    @SerializedName("itemLevel")
    @Expose
    var itemLevel: Int? = null
    @SerializedName("maxCount")
    @Expose
    var maxCount: Int? = null
    @SerializedName("maxDurability")
    @Expose
    var maxDurability: Int? = null
    @SerializedName("minFactionId")
    @Expose
    var minFactionId: Int? = null
    @SerializedName("minReputation")
    @Expose
    var minReputation: Int? = null
    @SerializedName("quality")
    @Expose
    var quality: Int? = null
    @SerializedName("sellPrice")
    @Expose
    var sellPrice: Int? = null
    @SerializedName("requiredSkill")
    @Expose
    var requiredSkill: Int? = null

    @SerializedName("requiredAbility")
    @Expose
    lateinit var requiredAbility: RequiredAbility

    @SerializedName("requiredLevel")
    @Expose
    var requiredLevel: Int? = null
    @SerializedName("requiredSkillRank")
    @Expose
    var requiredSkillRank: Int? = null

    @SerializedName("itemSource")
    @Expose
    lateinit var itemSource: ItemSource

    @SerializedName("baseArmor")
    @Expose
    var baseArmor: Int? = null
    @SerializedName("hasSockets")
    @Expose
    var hasSockets: Boolean? = null
    @SerializedName("isAuctionable")
    @Expose
    var isAuctionable: Boolean? = null
    @SerializedName("armor")
    @Expose
    var armor: Int? = null
    @SerializedName("displayInfoId")
    @Expose
    var displayInfoId: Int? = null

    @SerializedName("nameDescription")
    @Expose
    lateinit var nameDescription: String

    @SerializedName("nameDescriptionColor")
    @Expose
    lateinit var nameDescriptionColor: String

    @SerializedName("upgradable")
    @Expose
    var upgradable: Boolean? = null
    @SerializedName("heroicTooltip")
    @Expose
    var heroicTooltip: Boolean? = null

    @SerializedName("context")
    @Expose
    lateinit var context: String

    @SerializedName("bonusLists")
    @Expose
    var bonusLists: List<Any>? = null

    @SerializedName("availableContexts")
    @Expose
    var availableContexts: List<String>? = null

    @SerializedName("bonusSummary")
    @Expose
    lateinit var bonusSummary: BonusSummary

    @SerializedName("artifactId")
    @Expose
    var artifactId: Int? = null

}